components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1675,10 +1675,32 @@ bool RewardsServiceImpl::ShouldShowOnboarding() const {
 void RewardsServiceImpl::SaveOnboardingResult(OnboardingResult result) {
   PrefService* prefs = profile_->GetPrefs();
   prefs->SetTime(prefs::kOnboarded, base::Time::Now());
-  if (result == OnboardingResult::kOptedIn) {
-    SetAutoContributeEnabled(true);
-    SetAdsEnabled(true);
+
+  if (result != OnboardingResult::kOptedIn) {
+    return;
+  }
+
+  if (!IsInitialized()) {
+    StartProcess(base::BindOnce(
+        &RewardsServiceImpl::OnStartProcessForOnboarding, AsWeakPtr()));
+    return;
+  }
+
+  CompleteOnboarding();
+}
+
+void RewardsServiceImpl::OnStartProcessForOnboarding(
+    const ledger::type::Result result) {
+  if (result != ledger::type::Result::LEDGER_OK) {
+    return;
   }
+
+  CompleteOnboarding();
+}
+
+void RewardsServiceImpl::CompleteOnboarding() {
+  SetAutoContributeEnabled(true);
+  SetAdsEnabled(true);
```

Comment:
@zenparsing Is it safe to just enable ads like this? It seems like some regions don't support ads.

Reply:
Good catch - it looks like there are `ads_service->IsSupportedLocale(...)` checks in various places, but it's missing from `RewardsService::SetAdsEnabled`. Or maybe the check should be in `OnWalletCreatedForSetAdsEnabled`?

Reply:
It seems like everything boils down to `OnWalletCreatedForSetAdsEnabled`, so that would be my vote.

Reply:
That is correct in that some regions do not support ads

Reply:
Thanks @tmancey, I'll add the `ads_service->IsSupportedLocale(...)` check to `OnWalletCreatedForSetAdsEnabled`.